lib/web_ui/lib/src/engine/compositor/layer.dart

```diff
@@ -328,8 +328,58 @@ class PhysicalShapeLayer extends ContainerLayer
   @override
   void preroll(PrerollContext prerollContext, Matrix4 matrix) {
     prerollChildren(prerollContext, matrix);
-    paintBounds =
-        ElevationShadow.computeShadowRect(_path.getBounds(), _elevation);
+    if (_elevation == 0.0) {
+      paintBounds = _path.getBounds();
```

Comment:
nit: I think the function would be slightly more readable with an early return instead of if/else:

```
paintBounds = _path.getBounds();

if (_elevation == 0.0) {
  // Shadow does not contribute to paint bounds because there's no shadow.
  return;
}

... do the rest of the math ...
```

Reply:
Done.